src/window.ts

```diff
@@ -116,19 +119,22 @@ export type SEventsInfos = {
     nightlies: Array<number>;
     night: boolean;
   };
-} & {
-  // Christmas
-  holidayseason?: boolean;
-  grinch?: SMonsterEvent;
-  snowman?: SMonsterEvent;
-} & {
-  crabxx?: SMonsterEvent;
-};
+} & Partial<Record<EventKey, SMonsterEvent>> & {
+    // Christmas
+    holidayseason?: boolean; // unsure if this will override the above general purpose EventKey
+  } & Partial<Record<"grinch" | "snowman", SMonsterEvent>> & {
+    valentines?: boolean;
+  } & {
+    // halloween
+    halloween?: boolean; // unsure if this will override the above general purpose EventKey
+  } & Partial<Record<"mrpumpkin" | "mrgreen", SMonsterEvent>> & {
+    crabxx?: SMonsterEvent;
+  };
```

Comment:
I have changed this section a bit, you should probably rebase.

Reply:
I should have merged dev already, I solved a merge conflict at least :)

Reply:
Oh yeah, I didn't notice you had switched grinch and snowman into a `Record`! I think it's a bit harder to read for this specific case where we intersect multiple objects.

Reply:
🤔 I'll try and see if I can type it differently, I think my issue was that I could not access the value with a key, can't remember the exact error I had right now though. 

parent.S[key] would not work